.azure-pipelines/test_plan.py

```diff
@@ -992,7 +1004,23 @@ def poll(self, test_plan_id, interval=60, timeout=-1, expected_state="", expecte
             explicitly_specify_test_module = args.features or args.scripts
             if args.test_plan_type == "PR":
                 args.test_set = args.test_set if args.test_set else args.topology
-                parsed_script, parsed_specific_param = get_test_scripts(args.test_set)
+                incoming_test_scripts = []
+                if args.extend_test_set:
+                    res = subprocess.run(['git', 'diff', f'origin/{args.mgmt_branch}..HEAD'],
+                                         capture_output=True, text=True)
+                    print(f"Git diff: {res.stdout}")
+                    print(f"Git diff ERR: {res.stderr}")
+                    cmd = "git diff origin/{}..HEAD --name-only | grep \"tests/.*/test_.*\\.py$\"".format(
+                          args.mgmt_branch)
+                    print(f"CMD: {cmd}")
+                    res = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+                    incoming_test_scripts = str(res.stdout).split()
+                    for i in range(0, len(incoming_test_scripts)):
+                        # trim the "tests/" prefix
+                        incoming_test_scripts[i] = incoming_test_scripts[i][6:]
+
+                print(f"Incoming test scripts: {incoming_test_scripts}")
+                parsed_script, parsed_specific_param = get_test_scripts(args.test_set, incoming_test_scripts)
```

Comment:
Here, you want to add the incoming test scripts dynamically into PR testing, right?